java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksExternalClient.java

```diff
@@ -72,11 +72,38 @@
 
   private String apiKey = "";
 
-  public HopsworksExternalClient(String host, int port, Region region,
+  public HopsworksExternalClient(String host, int port, String apiKeyFilepath,
+                                 boolean hostnameVerification, String trustStorePath)
+          throws IOException, FeatureStoreException, KeyStoreException, CertificateException,
+          NoSuchAlgorithmException, KeyManagementException {
+    this(host, port, null, null, hostnameVerification, trustStorePath, apiKeyFilepath, null);
+  }
+
+  public HopsworksExternalClient(String host, int port, boolean hostnameVerification,
+                                 String trustStorePath, Region region, SecretStore secretStore)
+          throws IOException, FeatureStoreException, KeyStoreException, CertificateException,
+          NoSuchAlgorithmException, KeyManagementException {
+    this(host, port, region, secretStore, hostnameVerification, trustStorePath, null, null);
+  }
+
+
+  public HopsworksExternalClient(String host, int port, boolean hostnameVerification,
+                                 String trustStorePath, String apiKeyValue)
+          throws IOException, FeatureStoreException, KeyStoreException, CertificateException,
+          NoSuchAlgorithmException, KeyManagementException {
+    this(host, port, null, null, hostnameVerification, trustStorePath, null, apiKeyValue);
+  }
+
+  public HopsworksExternalClient(CloseableHttpClient httpClient, HttpHost httpHost) {
+    this.httpClient = httpClient;
+    this.httpHost = httpHost;
+  }
+
+  HopsworksExternalClient(String host, int port, Region region,
                                  SecretStore secretStore, boolean hostnameVerification,
-                                 String trustStorePath, String apiKeyFilepath)
-      throws IOException, FeatureStoreException, KeyStoreException, CertificateException,
-      NoSuchAlgorithmException, KeyManagementException {
+                                 String trustStorePath, String apiKeyFilepath, String apiKeyValue)
+          throws IOException, FeatureStoreException, KeyStoreException, CertificateException,
```

Comment:
it's strange that the checkstyle-plugin doesn't complain about the additional white space before the `throws`. but ok, then it should be fine

Reply:
If checkstyle doesn't complain, that is good enough for me. Is that ok?

Reply:
yes as long as checkstyle isn't complaining, it's fine